OAK-Marketplace-Emotion-Recognition/emotion_analyzer.py

```diff
@@ -0,0 +1,245 @@
+import json
+import cv2
+import matplotlib.pyplot as plt
+import numpy as np
+from PIL import Image
+import operator
+import os
+from modelplace_api.visualization import (WHITE_TEXT_COLOR, add_info, BACKGROUND_COLOR,
+                                          draw_emotion_recognition_one_frame)
+from utils.visualization import overlay_image, add_class_names_and_percents
+
+
+class EmotionAnalyzer:
+    def __init__(self, out_statistics_path: str = "result_statistic.json",
+                 visualization_size: int = 300,
+                 report_statistics_size: int = 720,
+                 emotion_images_path: str = "images/emoji",
+                 overlay_image_path: str = "images/overlay.png"):
+        self.y_offset_percent = 0.05
+        self.emotion_bar_offset_percent = 0.1
+
+        self.output_statistics_path = out_statistics_path
+        self.bar_size, self.padding, emotion_size = self._relative_size_estimation(visualization_size)
+        self.report_statistics_shape = report_statistics_size
+        self.overlay = cv2.resize(cv2.imread(overlay_image_path, cv2.IMREAD_UNCHANGED), (visualization_size, visualization_size))
+        self._init_bars(emotion_size=emotion_size, emotion_images_path=emotion_images_path)
+
+    def _init_bars(self, emotion_size, emotion_images_path):
+        self.bars = {
+            "happy": {"bar": EmotionBar(emotion='happy', bar_size=self.bar_size,
+                                        emotion_size=emotion_size,
+                                        emotion_path=os.path.join(emotion_images_path, 'happy.png')), "count": 0},
+            "sad": {"bar": EmotionBar(emotion='sad', bar_size=self.bar_size, emotion_size=emotion_size,
+                              emotion_path=os.path.join(emotion_images_path, 'sad.png')), "count": 0},
+            "anger": {"bar": EmotionBar(emotion='anger', bar_size=self.bar_size, emotion_size=emotion_size,
+                                emotion_path=os.path.join(emotion_images_path, 'anger.png')), "count": 0},
+            "surprise": {"bar": EmotionBar(emotion='surprise', bar_size=self.bar_size, emotion_size=emotion_size,
+                                   emotion_path=os.path.join(emotion_images_path, 'surprise.png')), "count": 0},
+            "neutral": {"bar":EmotionBar(emotion='neutral', bar_size=self.bar_size, emotion_size=emotion_size,
+                                  emotion_path=os.path.join(emotion_images_path, 'neutral.png')), "count": 0}
+        }
+        self._fill_bars()
+
+    def _fill_bars(self):
+        for x in self.bars.values():
+            x["bar"].draw()
+
+    def _increment_count(self, emotion):
+        self.bars[emotion]["count"] += 1
+
+    def _relative_size_estimation(self, visualization_size, bar_amount: int = 5):
```

Comment:
Bar amount can be easily calculated using `len(self.bars)`

Reply:
When we generate result statistic image with bars, we have actually five bars, but we need to place only four in line

Reply:
What do you think about usage `len(self.bars) - 1` then?

Reply:
Discussed